internal/core/items/flickable.rs

```diff
@@ -241,19 +240,23 @@ impl FlickableData {
                         if crate::animations::current_tick() - pressed_time > DURATION_THRESHOLD {
                             return false;
                         }
-                        let can_move_horiz = (Flickable::FIELD_OFFSETS.viewport
-                            + Empty::FIELD_OFFSETS.width)
-                            .apply_pin(flick)
-                            .get()
-                            > flick.width();
-                        let can_move_vert = (Flickable::FIELD_OFFSETS.viewport
-                            + Empty::FIELD_OFFSETS.height)
-                            .apply_pin(flick)
-                            .get()
-                            > flick.height();
+                        // Check if the mouse was moved more than the DISTANCE_THRESHEOLD in a
+                        // direction in which the flickable can flick
                         let diff = position - inner.pressed_pos;
-                        (can_move_horiz && diff.x.abs() > DISTANCE_THRESHOLD)
-                            || (can_move_vert && diff.y.abs() > DISTANCE_THRESHOLD)
+                        let vp = Flickable::FIELD_OFFSETS.viewport;
+                        let gap_h = (vp + Empty::FIELD_OFFSETS.width).apply_pin(flick).get()
+                            - flick.width();
+                        let gap_v = (vp + Empty::FIELD_OFFSETS.height).apply_pin(flick).get()
+                            - flick.height();
+                        (gap_h > LogicalLength::default() && {
+                            let x = (vp + Empty::FIELD_OFFSETS.x).apply_pin(flick).get();
+                            (x < LogicalLength::default() && diff.x_length() > DISTANCE_THRESHOLD)
+                                || (x > -gap_h && diff.x_length() < -DISTANCE_THRESHOLD)
+                        }) || (gap_v > LogicalLength::default() && {
+                            let y = (vp + Empty::FIELD_OFFSETS.y).apply_pin(flick).get();
+                            (y < LogicalLength::default() && diff.y_length() > DISTANCE_THRESHOLD)
+                                || (y > -gap_v && diff.y_length() < -DISTANCE_THRESHOLD)
+                        })
```

Comment:
At this point it's hard to see what the boolean result of this super long expression means. I suggest to split it up a little - unless this can be parametrised somehow (not sure if worth it).

```suggestion
                        let intercept_horizontal = || {
                            gap_h > LogicalLength::default() && {
                                let x = (vp + Empty::FIELD_OFFSETS.x).apply_pin(flick).get();
                                (x < LogicalLength::default()
                                    && diff.x_length() > DISTANCE_THRESHOLD)
                                    || (x > -gap_h && diff.x_length() < -DISTANCE_THRESHOLD)
                            }
                        };
                        let intercept_vertical = || {
                            gap_v > LogicalLength::default() && {
                                let y = (vp + Empty::FIELD_OFFSETS.y).apply_pin(flick).get();
                                (y < LogicalLength::default()
                                    && diff.y_length() > DISTANCE_THRESHOLD)
                                    || (y > -gap_v && diff.y_length() < -DISTANCE_THRESHOLD)
                            }
                        };
                        intercept_horizontal() || intercept_vertical()
```